src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -102,14 +102,16 @@ def _compute_mask_indices(
         if sz - min_len <= num_mask:
             min_len = sz - num_mask - 1
 
-        mask_idc = np.random.choice(sz - min_len, num_mask, replace=False)
-        mask_idc = np.asarray([mask_idc[j] + offset for j in range(len(mask_idc)) for offset in range(lengths[j])])
-        mask_idcs.append(np.unique(mask_idc[mask_idc < sz]))
+        mask_idc = torch.randperm(sz - min_len)[:num_mask]
+        mask_idc = torch.from_numpy(
+            np.asarray([mask_idc[j] + offset for j in range(len(mask_idc)) for offset in range(lengths[j])])
```

Comment:
Let's try to not use `np` here. If we use `torch.from_numpy(...)` - it's not GPU-friendly

Reply:
mask_idc = torch.tensor([mask_idc[j] + offset for j in range(len(mask_idc)) for offset in range(lengths[j])])
 
 I just did this. Is this right ?